engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/Instrumentor.java

```diff
@@ -61,48 +58,48 @@ final Instrumentor deactivate() {
   // Callbacks
   //
   @Override
-  public Object findCachedResult(UUID nodeId) {
+  public Object findCachedResult(IdExecutionService.Info info) {
     try {
       if (onEnter != null) {
-        var ret = InteropLibrary.getUncached().execute(onEnter, nodeId.toString());
+        var ret = InteropLibrary.getUncached().execute(onEnter, info.getId().toString());
         ret = InteropLibrary.getUncached().isNull(ret) ? null : ret;
-        return handle.isDisposed() ? null : ret;
-      }
-    } catch (InteropException ex) {
+        return handle.isDisposed() ? null : ret;    }
+    } catch (InteropException ignored) {
     }
     return null;
   }
 
   @Override
-  public void updateCachedResult(UUID nodeId, Object result, boolean isPanic, long nanoElapsedTime) {
+  public void updateCachedResult(IdExecutionService.Info info) {
     try {
       if (onReturn != null) {
-        InteropLibrary.getUncached().execute(onReturn, nodeId.toString(), result);
+        InteropLibrary.getUncached().execute(onReturn, info.getId().toString(), info.getResult());
```

Comment:
Can we also expose the `eval` functionality from `Instrumentor` API, please? For example change the [following line](https://github.com/enso-org/enso/blob/f47ef0c11144f03ce045da9732a9e45358431b8a/distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso#L625) to:
```
n_return self (fn : Text -> Any -> Nothing) expression=Nothing =
```
e.g. be also able to specify `expression` to be evaluated in the context of executing program.

Reply:
Extracted into #8293